File: trunk/download/stream/src/main/java/org/simpleframework/xml/core/AttributeException.java
/*
 * AttributeException.java July 2006
 *
 * Copyright (C) 2006, Niall Gallagher <niallg@users.sf.net>
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the 
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General 
 * Public License along with this library; if not, write to the 
 * Free Software Foundation, Inc., 59 Temple Place, Suite 330, 
 * Boston, MA  02111-1307  USA
 */

package org.simpleframework.xml.core;

/**
 * The <code>AttributeException</code> is used to represent conditions 
 * when an XML attribute is in an invalid state. Typically this is 
 * used when and attribute cannot be serialized or deserialized. Such 
 * a condition can arise when an attribute exists within the source 
 * XML that does not match the schema class. For serialization this
 * exception is thrown if a required field is null.
 * 
 * @author Niall Gallagher
 */
public class AttributeException extends PersistenceException {

   /**
    * Constructor for the <code>AttributeException</code> object. This
    * constructor takes a format string an a variable number of object
    * arguments, which can be inserted into the format string. 
    * 
    * @param text a format string used to present the error message
    * @param list a list of arguments to insert into the string
    */
   public AttributeException(String text, Object... list) {
      super(text, list);           
   }        

   /**
    * Constructor for the <code>AttributeException</code> object. This
    * constructor takes a format string an a variable number of object
    * arguments, which can be inserted into the format string. 
    * 
    * @param cause the source exception this is used to represent
    * @param text a format string used to present the error message
    * @param list a list of arguments to insert into the string 
    */
   public AttributeException(Throwable cause, String text, Object... list) {
      super(cause, text, list);           
   }
}

File: trunk/download/stream/port/src/main/Xml/Core/AttributeException.cs
#region License
//
// AttributeException.cs July 2006
//
// Copyright (C) 2006, Niall Gallagher <niallg@users.sf.net>
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
// implied. See the License for the specific language governing
// permissions and limitations under the License.
//
#endregion
#region Using directives
using System;
#endregion
namespace SimpleFramework.Xml.Core {
   /// <summary>
   /// The <c>AttributeException</c> is used to represent conditions
   /// when an XML attribute is in an invalid state. Typically this is
   /// used when and attribute cannot be serialized or deserialized. Such
   /// a condition can arise when an attribute exists within the source
   /// XML that does not match the schema class. For serialization this
   /// exception is thrown if a required field is null.
   /// </summary>
   public class AttributeException : PersistenceException {
      /// <summary>
      /// Constructor for the <c>AttributeException</c> object. This
      /// constructor takes a format string an a variable number of object
      /// arguments, which can be inserted into the format string.
      /// </summary>
      /// <param name="text">
      /// a format string used to present the error message
      /// </param>
      /// <param name="list">
      /// a list of arguments to insert into the string
      /// </param>
      public AttributeException(String text, Object... list) {
         super(text, list);
      }
      /// <summary>
      /// Constructor for the <c>AttributeException</c> object. This
      /// constructor takes a format string an a variable number of object
      /// arguments, which can be inserted into the format string.
      /// </summary>
      /// <param name="cause">
      /// the source exception this is used to represent
      /// </param>
      /// <param name="text">
      /// a format string used to present the error message
      /// </param>
      /// <param name="list">
      /// a list of arguments to insert into the string
      /// </param>
      public AttributeException(Throwable cause, String text, Object... list) {
         super(cause, text, list);
      }
   }
}
